go/obscuronode/host/host.go

```diff
@@ -291,7 +291,18 @@ func (a *Node) startProcessing() {
 	a.EnclaveClient.Start(lastBlock)
 
 	if a.config.IsGenesis {
-		a.initialiseProtocol(&lastBlock)
+		// if the ContractMgmtBlkHash is not specified (like in the ethereum_mock)
+		// the genesis rollup will use the latest block
+		if a.config.ContractMgmtBlkHash.Hex() == (common.Hash{}).Hex() {
+			a.initialiseProtocol(lastBlock.Hash())
+		} else {
+			// the block should be available in the l1
+			blk, err := a.ethClient.BlockByHash(*a.config.ContractMgmtBlkHash)
+			if err != nil {
+				panic(err)
```

Comment:
This error should indicate that it's the contract mgmt block in particular that's missing. Otherwise, it will be a generic "block missing" error.